### PR TITLE
chore: add waitress for win32 platform

### LINKED ISSUE #22918: can't find '_main_' module in 'superset'
Hello @eschutho  @craig-rueda  @hughhhh @michael-s-molina 
When I want to run the Superset project directly from the github project instead of installing it in virtualenv, I get the following error.
Error Description: can't find '_main_' module in 'superset'

### LINKED ISSUE #22922: superset runtime error
We want to run superset with waitress instead of gunicorn on windows like this : waitress-serve --host 127.0.0.1 superset:app
our goal is to run Superset project directly from github project instead of installing it in virtualenv.
but we are getting this error, any idea what to do? 
Error : raise RuntimeError(_app_ctx_err_msg)
RuntimeError: Working outside of application context.

### PR DESCRIPTION
**Pull request by EugeneTorap**
### SUMMARY
This PR add a platform condition while installing libs from `setup.py`.
If sys platform is Windows we ignore `gunicorn` installation and instead of it we install `waitress`

### ADDITIONAL INFORMATION

- [x] Has associated issue:
Fixes #22922
Fixes #22918
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by EugeneTorap** [comment]
Hey @rusackas @sedanursayer @zeynepkoyun!
I guess lib installation should be work good.
Can you test the logic locally on Windows machine?

**Comment by zeynepkoyun** [comment]
Hello, we added the relevant line of code to the setup.py file, but it didn't work.

**Comment by EugeneTorap** [comment]
Hi @zeynepkoyun. Do you run `pip install -e .` for it?

**Comment by zeynepkoyun** [comment]
yes, after editing setup.py, I installed it, but superset app was started with waitress-serve

**Comment by EugeneTorap** [comment]
How do you run superset app?

**Comment by zeynepkoyun** [comment]
waitress-serve --host 0.0.0.0:8088 superset:app

**Comment by EugeneTorap** [comment]
`flask run -p 8080`
Does it work?

**Comment by zeynepkoyun** [comment]
I'll test it and let you know

**Comment by EugeneTorap** [comment]
For waitress you should use the next command:
```
waitress-serve --host 127.0.0.1 --call superset:create_app
```

**Comment by EugeneTorap** [comment]
@zeynepkoyun About `country_holidays` issue. Can you text me in Slack? It will be more efficient

**Comment by zeynepkoyun** [comment]
How can I reach you on slack?

**Comment by EugeneTorap** [comment]
I have the same nickname in Slack

**Comment by EugeneTorap** [comment]
@zeynepkoyun http://bit.ly/join-superset-slack

**Comment by EugeneTorap** [comment]
@villebro @dpgaspar Can you review it?

File: setup.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def get_git_sha() -> str:
         "func_timeout",
         "geopy",
         "graphlib-backport",
-        "gunicorn>=20.1.0",
+        "gunicorn>=20.1.0; sys_platform != 'win32'",
```
Comment (craig-rueda) [inline setup.py:96]: Please make sure you run `pip-compile` after updating this: https://github.com/apache/superset/blob/master/CONTRIBUTING.md#dependencies

Comment (EugeneTorap) [reply setup.py:96]: @craig-rueda I run pip-compile-multi on Mac M1 and nothing

Comment (craig-rueda) [reply setup.py:96]: Ok, that's probably expected

**Review by craig-rueda — APPROVED** [review]

**Pull request merged**